metrics/web/test/dashboard/presentation/state/project_metrics_notifier_test.dart

```diff
@@ -227,6 +257,96 @@ void main() {
       expect(performancePoints.map((p) => p.y), equals(expectedY));
     });
 
+    test(
+      "maps the finished build results to finished build result view models",
+      () async {
+        final dashboardMetrics = DashboardProjectMetrics(
+          projectId: 'id',
+          buildResultMetrics: BuildResultMetric(
+            buildResults: [
+              createBuildResult(BuildStatus.successful),
+              createBuildResult(BuildStatus.unknown),
+              createBuildResult(BuildStatus.failed),
+            ],
+          ),
+        );
+        when(receiveProjectMetricsMock.call(any)).thenAnswer(
+          (_) => Stream.value(dashboardMetrics),
+        );
+
+        final notifier = ProjectMetricsNotifier(receiveProjectMetricsMock);
+
+        await setUpNotifier(notifier: notifier, projects: projects);
+
+        final projectMetrics = notifier.projectsMetricsTileViewModels.first;
+        final buildResultMetrics = projectMetrics.buildResultMetrics;
+
+        final viewModels = buildResultMetrics.buildResults;
+
+        expect(viewModels, everyElement(isA<FinishedBuildResultViewModel>()));
+      },
+    );
+
+    test(
+      "maps the in-progress build results to in progress build result view models",
+      () async {
+        final dashboardMetrics = DashboardProjectMetrics(
+          projectId: 'id',
+          buildResultMetrics: BuildResultMetric(
+            buildResults: [createBuildResult(BuildStatus.inProgress)],
+          ),
+        );
+        when(receiveProjectMetricsMock.call(any)).thenAnswer(
+          (_) => Stream.value(dashboardMetrics),
+        );
+
+        final notifier = ProjectMetricsNotifier(receiveProjectMetricsMock);
+
+        await setUpNotifier(notifier: notifier, projects: projects);
+
+        final projectMetrics = notifier.projectsMetricsTileViewModels.first;
+        final buildResultMetrics = projectMetrics.buildResultMetrics;
+
+        final viewModels = buildResultMetrics.buildResults;
+
+        expect(viewModels, isNotEmpty);
+        expect(viewModels, everyElement(isA<InProgressBuildResultViewModel>()));
+      },
+    );
+
+    test(
+      "loads a build result metric with the maximum build duration from the build results",
+      () async {
+        const expectedMaximumDuration = Duration(days: 3);
+        final dashboardMetrics = DashboardProjectMetrics(
+          projectId: 'id',
+          buildResultMetrics: BuildResultMetric(
+            buildResults: [
```

Comment:
Would it be great to have a couple of builds here to ensure that the `ProjectMetricsNotifier` chooses the biggest one?